scripts/modules/b2b-account/quotes.js

```diff
@@ -159,25 +159,50 @@ define([
                 var dateValue =  $(this).val();
                 self.filterGrid(nameValue, dateValue, collection);
             });
-            
+            if (viewB2BAccount) {
+                collection.filterBy("customerAccountId eq " + self.model.attributes.accountToView);
+            }
+           
             this.initializeGrid(collection);
         },
         filterGrid: function (nameValue, dateValue, collection) {
-            filterstring = "";
+            var filterstring = "";
+            var self = this;
+            var viewB2BAccount = self.model.attributes.viewB2BAccount;
+            if (viewB2BAccount) {
+                filterstring = "customerAccountId eq " + self.model.attributes.accountToView;
+            }
+           
             if (nameValue !== "") {
                 nameValue = nameFilter + nameValue;
-                filterstring = nameValue;
+                if (filterstring === "") {
```

Comment:
Ideally you should refactor this if/else logic into a helper function called `addFilterParameter` which will take a value like `customerAccountId eq 1234` and add it to a filterString var you pass in. It's not strictly necessary but would be cleaner.

Reply:
OK that's fine.